pkg/api/validation/validation.go

```diff
@@ -30,12 +30,32 @@ import (
 	"github.com/golang/glog"
 )
 
+const minPort int = 1
+const minHostPort int = 0
+const maxPort int = 65536
+
+const QualifiedNameErrorMsg string = "must match regex [" + util.Dns1123SubdomainFmt + " / ] " + util.Dns1123LabelFmt
+const PortNameErrorMsg string = "must be less than %d characters and match regex " + util.Dns1123LabelFmt
+const CIdentifierErrorMsg string = "must match regex " + util.CIdentifierFmt
+const NegativeErrorMsg string = "value cannot be negative"
+
+func intervalErrorMsg(lo, hi int) string {
+	return fmt.Sprintf("must be greater than %d and less than %d", lo, hi)
+}
+
+var DnsSubdomainErrorMsg string = "must match regex " + util.Dns1123SubdomainFmt + " and have at most " + fmt.Sprintf("%d", util.Dns1123SubdomainMaxLength) + "characters"
+var DnsLabelErrorMsg string = "must match regex " + util.Dns1123LabelFmt + " and have at most " + fmt.Sprintf("%d", util.Dns1123LabelMaxLength) + "characters"
+var DNS952LabelErrorMsg string = "must match regex " + util.Dns952LabelFmt + " and have at most " + fmt.Sprintf("%d", util.Dns952LabelMaxLength) + "characters"
+var PdPartitionErrorMsg string = intervalErrorMsg(0, 255)
+var PortErrorMsg string = intervalErrorMsg(minPort, maxPort)
+var HostPortErrorMsg string = intervalErrorMsg(minHostPort, maxPort)
```

Comment:
I think HostPortErrorMsg can go away and just use PortRangeErrorMsg - 0 is handled specially in the validation, and it should not be an allowed value (but that is how Go does things)